internal/pkg/manifest/validate.go

```diff
@@ -1937,36 +1948,54 @@ func populateNLBPortsAndValidate(containerNameFor map[uint16]string, opts valida
 		return nil
 	}
 	nlb := opts.nlb
-	if nlb.Port == nil {
+	if nlb.MainListener.Port == nil {
 		return nil
 	}
-	nlbPort, _, err := ParsePortMapping(nlb.Port)
+
+	for _, listener := range nlb.NLBListeners() {
+		if err := populateAndValidateNLBPorts(listener, containerNameFor, opts); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func populateAndValidateNLBPorts(listener NetworkLoadBalancerListener, containerNameFor map[uint16]string, opts validateExposedPortsOpts) error {
```

Comment:
Do we still need the `opts validateExposedPortsOpts` parameter?

Reply:
I think we can just need `mainContainerName`, so we can just send that param instead. Made the changes.